### PR TITLE
Closes #6 and #7: Docs changes for an easier start, and to remove reliance on Ninja

### LINKED ISSUE #6: sdfx_st/examples/README.md doesn't say possible values for MBED_TOOLCHAIN=<TOOLCHAIN>
User story: As a new user, I'm trying to follow https://github.com/MCU-Driver-HAL/MCU-Driver-ST/blob/main/sdfx_st/examples/README.md . However, I get stuck at this step:

```
cmake -S . -B cmake_build -GNinja -DCMAKE_BUILD_TYPE=debug -DMBED_TOOLCHAIN=<TOOLCHAIN>
```

I have no idea what values for `<TOOLCHAIN>` are possible. (And related, which one of supposedly many alternatives I'd be recommended to use for quick start.)

### PR DESCRIPTION
**Pull request by Jambls**
Changed the docs in the examples and test folders to:
Explain the options for the toolchain, and link back to the getting started doc for more information.
Get rid of using ninja as the default, and include some information to help if they're using a non standard build system.

Made the readme slightly clearer.

One of the links in here (theoretically) links to a page only created in the sibling PR on the MCU-Driver-HAL repo

### CODE REVIEW

**Review by Gallagator — COMMENTED** [review]
I would suggest that you define what \<INTERFACE\> means. I don't think its been defined in this file.

File: sdfx_st/tests/README.md [hunk 1]
```diff
@@ -14,14 +14,15 @@ Each test can be built by doing the following:
...
+1. Change the current working directory to `MCU-Driver-ST/sdfx_st/tests/mbed_hal/<TEST>/` with `<TEST>` being the test that you would like to run, eg `echo`.
 1. Generate the build system files:
 
     ```
-    cmake -S ./ -B cmake_build/ -GNinja -DGREENTEA_CLIENT_STDIO=OFF -DMBED_TOOLCHAIN=<TOOLCHAIN> -DCMAKE_BUILD_TYPE=debug
+    cmake -S ./ -B cmake_build/ -DGREENTEA_CLIENT_STDIO=OFF -DMBED_TOOLCHAIN=<TOOLCHAIN> -DCMAKE_BUILD_TYPE=debug
     ```
 
-    If you prefer to use UNIX Makefiles rather than the Ninja build system, remove `-GNinja` from the command above.
+    If you are using a non-default built system, and you have not set the environment variable, then you can append `-G<BuildSystem>` to the command above.
```
Comment (Gallagator) [inline sdfx_st/tests/README.md:24]: Minor typo. built should be build.

File: sdfx_st/examples/README.md [hunk 2]
```diff
@@ -11,17 +11,19 @@ Each application can be built by doing the following:
     ```
     git clone --recursive https://github.com/MCU-Driver-HAL/MCU-Driver-ST
     ```
-1. Change the current working directory to `MCU-Driver-ST/sdfx_st/examples/<INTERFACE>/`
+1. Change the current working directory to `MCU-Driver-ST/sdfx_st/examples/<INTERFACE>/` with `<INTERFACE>` being the example you want to run, eg `gpio`.
```
Comment (MarceloSalazar) [inline sdfx_st/examples/README.md:14]:
```suggestion
1. Change the current working directory to `MCU-Driver-ST/sdfx_st/examples/<INTERFACE>/` with `<INTERFACE>` being the example for the API you want to run, eg `gpio`.
```

File: sdfx_st/examples/README.md [hunk 3]
```diff
@@ -11,17 +11,19 @@ Each application can be built by doing the following:
...
 1. Run:
     ```
-    cmake -S . -B cmake_build -GNinja -DCMAKE_BUILD_TYPE=debug -DMBED_TOOLCHAIN=<TOOLCHAIN>
+    cmake -S . -B cmake_build -DCMAKE_BUILD_TYPE=debug -DMBED_TOOLCHAIN=<TOOLCHAIN>
     ```
+    If you are using a non-default build system, and you have not set the environment variable, then you can append `-G<BuildSystem>` to the command above.
+    Possible values for `<TOOLCHAIN>` are `ARM` and `GCC-ARM`. More on that [here.](https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/blob/main/docs/getting_started/README.md#Compiler)
```
Comment (MarceloSalazar) [inline sdfx_st/examples/README.md:20]: Typo in GCC toolchain?
```suggestion
    Possible values for `<TOOLCHAIN>` are `ARM` and `GCC_ARM`. More on that [here.](https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/blob/main/docs/getting_started/README.md#Compiler)
```

File: sdfx_st/tests/README.md [hunk 4]
```diff
@@ -14,14 +14,15 @@ Each test can be built by doing the following:
...
 1. Generate the build system files:
 
     ```
-    cmake -S ./ -B cmake_build/ -GNinja -DGREENTEA_CLIENT_STDIO=OFF -DMBED_TOOLCHAIN=<TOOLCHAIN> -DCMAKE_BUILD_TYPE=debug
+    cmake -S ./ -B cmake_build/ -DGREENTEA_CLIENT_STDIO=OFF -DMBED_TOOLCHAIN=<TOOLCHAIN> -DCMAKE_BUILD_TYPE=debug
     ```
 
-    If you prefer to use UNIX Makefiles rather than the Ninja build system, remove `-GNinja` from the command above.
+    If you are using a non-default build system, and you have not set the environment variable, then you can append `-G<BuildSystem>` to the command above.
+    Possible values for `<TOOLCHAIN>` are `ARM` and `GCC-ARM`. More on that [here.](https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/blob/main/docs/getting_started/README.md#Compiler)
```
Comment (MarceloSalazar) [inline sdfx_st/tests/README.md:25]: Typo in GCC toolchain?
```suggestion
    Possible values for `<TOOLCHAIN>` are `ARM` and `GCC_ARM`. More on that [here.](https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/blob/main/docs/getting_started/README.md#Compiler)
```

File: README.md [hunk 5]
```diff
@@ -84,4 +84,3 @@ This repository contains STM32 families directories for STM32L475xG Part Number.
 ## License and contributions
 
 The software is provided under the [Apache-2.0 license](LICENSE-apache-2.0.txt). All contributions to software and documents are licensed by contributors under the same license model as the software/document itself (ie. inbound == outbound licensing). The MCU-Driver-HAL project may reuse software already licensed under another license, provided the license is permissive in nature and compatible with Apache v2.0.
-
```
Comment (MarceloSalazar) [inline README.md:87]: Please undo the change

File: sdfx_st/examples/README.md [hunk 6]
```diff
@@ -11,17 +11,19 @@ Each application can be built by doing the following:
...
     git clone --recursive https://github.com/MCU-Driver-HAL/MCU-Driver-ST
     ```
-1. Change the current working directory to `MCU-Driver-ST/sdfx_st/examples/<INTERFACE>/`
+1. Change the current working directory to `MCU-Driver-ST/sdfx_st/examples/<INTERFACE>/` with `<INTERFACE>` being the example for the API you want to run, eg `gpio`.
 1. Run:
     ```
     cmake -S . -B cmake_build -GNinja -DCMAKE_BUILD_TYPE=debug -DMBED_TOOLCHAIN=<TOOLCHAIN>
     ```
+    If you are using a non-default build system, and you have not set the environment variable, then you can append `-G<BuildSystem>` to the command above.
+    Possible values for `<TOOLCHAIN>` are `ARM` and `GCC_ARM`. More on that [here.](https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/blob/main/docs/getting_started/README.md#Compiler)
```
Comment (MarceloSalazar) [inline sdfx_st/examples/README.md:20]: Please check link - it doesn't seem to work

**Review by MarceloSalazar — APPROVED** [review]
